### DISCUSSION TITLE
Feature Request: Support for CodeceptJS Integration

### DISCUSSION THREAD

**Discussion by basal-john** [ortoniKC/ortoni-report#42] [Ideas]
Hello Koushik,

First, I want to commend you on developing Ortoni Report—it’s a comprehensive and visually appealing HTML report generator that has significantly enhanced our Playwright test reporting.

As a Quality Engineering Manager overseeing end-to-end test automation, we utilize CodeceptJS with Playwright and TypeScript for our testing framework. Currently, Ortoni Report is tailored specifically for Playwright tests. Integrating support for CodeceptJS would be highly beneficial for teams like ours that leverage this framework.

Benefits of CodeceptJS Integration:
	•	Unified Reporting: Consolidating test reports from both Playwright and CodeceptJS into a single, cohesive format would streamline our testing processes.
	•	Enhanced Test Management: Utilizing Ortoni Report’s features—such as hierarchical grouping, detailed test breakdowns, and customizable themes—would improve the management and analysis of our CodeceptJS test results.
	•	Broader Adoption: Supporting CodeceptJS could attract a wider user base to Ortoni Report, as many organizations employ this framework for their testing needs.

Potential Use Cases:
	•	Cross-Framework Testing: Teams conducting tests across multiple frameworks could benefit from a unified reporting tool, reducing the need for multiple reporting solutions.
	•	Consistent Reporting Standards: Organizations aiming for standardized reporting across different testing frameworks would find this integration advantageous.

Desired Features:
	•	Seamless Integration: The ability to generate reports for CodeceptJS tests with minimal configuration changes.
	•	Feature Parity: Ensuring that features available for Playwright tests, such as detailed test breakdowns and customizable themes, are also accessible for CodeceptJS tests.
	•	Compatibility: Support for various CodeceptJS helpers and plugins to accommodate diverse testing scenarios.

I believe that incorporating CodeceptJS support into Ortoni Report would greatly enhance its versatility and appeal to a broader audience. If there’s any way I can assist in this endeavor, please let me know.

Thank you for considering this feature request.

Regards,
Basal

**Comment by basal-john** [comment]
@ortoniKC : I would love to hear your thoughts on this :)

**Reply by ortoniKC** [reply]
Hi, I'm not sure if codecept has exposed their report API, and moreover it's not fully free or opensource so it might not be worthy.

**Comment by basal-john** [comment]
@ortoniKC : Thanks for your reply. CodeceptJS already supports many [reporters](https://codecept.io/reports/#reporters). This community driver project is completely open-sourced and free: https://github.com/codeceptjs/CodeceptJS. What feature made you think that this is not free ?

**Reply by ortoniKC** [reply]
Thanks for the clarification, I'll take a look

**Reply by basal-john** [reply]
@ortoniKC: Thank you. CodeceptJS has an average weekly download of 145,000. If you could add the compatibility, this would help a lot of us :)
